diff --git a/installwatch/create-localdecls b/installwatch/create-localdecls
index 0cb3e27..407e755 100755
--- a/installwatch/create-localdecls
+++ b/installwatch/create-localdecls
@@ -109,9 +109,9 @@ if test "$VERSION" = 'libc.so.6' ; then
 			echo '#define GLIBC_MINOR 13' >> $OUTFILE
 			SUBVERSION='glibc-2.13' ;;
 		*)
-			echo 'Treated as glibc >= 2.1 (finger crossed)'
-			echo '#define GLIBC_MINOR 1' >> $OUTFILE
-			SUBVERSION='glibc-2.1' ;;
+			echo 'Treated as glibc >= 2.13 (finger crossed)'
+			echo '#define GLIBC_MINOR 13' >> $OUTFILE
+			SUBVERSION='glibc-2.13' ;;
 	        esac
 		;;
 	esac
