diff --git a/Makefile b/Makefile
index 72ccf9d..58bb408 100644
--- a/Makefile
+++ b/Makefile
@@ -39,22 +39,8 @@ install: all
 	
 	mkdir -p $(CONFDIR)
 	install -m644  checkinstallrc-dist $(CONFDIR)
-	@if ! [ -f $(CONFDIR)/checkinstallrc ]; then \
-		cp $(CONFDIR)/checkinstallrc-dist $(CONFDIR)/checkinstallrc; \
-	else \
-		echo; \
-		echo; \
-		echo ======================================================== ;\
-		echo; \
-		echo An existing checkinstallrc file has been found. ;\
-		echo The one from this distribution can be found at: ; \
-		echo; \
-		echo -e \\t$(CONFDIR)/checkinstallrc-dist ; \
-		echo; \
-		echo; \
-		echo ======================================================== ;\
-		echo; \
-	fi
+	cp -p checkinstallrc-dist checkinstallrc
+	install -m644  checkinstallrc $(CONFDIR)
 	
 clean:
 	for file in locale/checkinstall-*.mo checkinstall ; do \
